### PR TITLE
Add user search feature to users dashboard

### LINKED ISSUE #156: UserList search feature
**Given** I am on the `/users` page
**Then** I should be able to search for specific users

### PR DESCRIPTION
**Pull request by bryszard**
Given I am on the /users page
Then I should be able to search for specific users

#### Issue addressed
Fixes #156

### CODE REVIEW

**Comment by mattwr18** [comment]
this is looking great @bryszard... I thought it would take longer to get the functionality working!
Screenshot at 2019-03-23 16:23:59

**Review by mattwr18 — CHANGES_REQUESTED** [review]
It's looking great @bryszard, as there is not `[WIP]`, I thought I'd give a review and ask if you could add cypress/component tests?

**Comment by bryszard** [comment]
Thank you @mattwr18 for the comment. In fact I left it as `[WIP]`, because:
1. There are no specs.
2. I was not sure to the extent of searching we want to have here. Right now we're just querying the first_name + last_name. But on the old website there is another approach (although not working currently) - we have a dropdown with the category we want to query. So we could choose some specific attributes from ones available for user, eg. slug, projects etc. and then we'd query those instead of just first_name + last_name.

What do you think? What should be the extent of the search options here?

EDIT: Ah, I see that the bot created the 'review' label automatically. I don't really know how it works right now.

File: src/containers/UsersList.js [hunk 1]
```diff
@@ -68,6 +68,16 @@ export class UsersList extends Component {
     })
   };
 
+  handleSearchInput = e => {
+    const users = this.props.users.filter(user => {
+      const name = `${user.first_name} ${user.last_name}`
+
+      return name.includes(e.target.value)
```
Comment (arku) [inline src/containers/UsersList.js]: What do you think of sanitizing the user input (trimming and lowercasing it) and lowercasing the user name?

File: src/containers/UsersList.js [hunk 2]
```diff
@@ -83,12 +93,23 @@ export class UsersList extends Component {
...
+                  icon='search'
+                  placeholder='Search...'
+                />
                 <Card.Group centered itemsPerRow={3}>
-                  <Paginate
-                    items={usersList}
-                    Component={User}
-                    pageCount={pageCount}
-                  />
+                  {usersList ? (
```
Comment (arku) [inline src/containers/UsersList.js:109]: I might be missing something, is the `userList` check enough to check if there are more than one page of user objects?

**Review by arku — COMMENTED** [review]
LGTM. Left a couple of minor comments.

Take a look when you get a chance.

File: src/containers/UsersList.js (hunk 2 above)
Comment (mattwr18) [reply src/containers/UsersList.js:109]: at the moment @arku, @bryszard we are using this in the `ProjectsList` to not display the `PaginationLinks` if their is only one page, I believe https://github.com/AgileVentures/WebsiteOne-FE/blob/develop/src/containers/ProjectsList.js#L132

maybe you could try that out @bryszard when you get a chance?

**Comment by mattwr18** [comment]
> Thank you @mattwr18 for the comment. In fact I left it as `[WIP]`, because:
> 
>     1. There are no specs.
> 
>     2. I was not sure to the extent of searching we want to have here. Right now we're just querying the first_name + last_name. But on the old website there is another approach (although not working currently) - we have a dropdown with the category we want to query. So we could choose some specific attributes from ones available for user, eg. slug, projects etc. and then we'd query those instead of just first_name + last_name.
> 
> 
> What do you think? What should be the extent of the search options here?
> 
> EDIT: Ah, I see that the bot created the 'review' label automatically. I don't really know how it works right now.

I added a `[WIP]` tag to the title so it sticks out more... I think we could have more complicated searching, but maybe we could get this reviewed and merged in, after there are tests, first, and then take on other use cases? 
what do you think @joaopapereira?

File: src/containers/UsersList.js (hunk 1 above)
Comment (bryszard) [reply src/containers/UsersList.js]: Will do, thanks.

File: src/containers/UsersList.js (hunk 2 above)
Comment (bryszard) [reply src/containers/UsersList.js:109]: Thanks for that, I will add it.

**Comment by bryszard** [comment]
@mattwr18 @arku I've added some unit tests for jest and refactored as you indicated. I can add more unit tests if you want. I guess also cypress spec update is missing, but I don't know how to use cypress yet.

**Comment by mattwr18** [comment]
@bryszard would you be interested in pairing on the cypress tests?

**Comment by bryszard** [comment]
Now it should be complete.

File: src/containers/UsersList.js [hunk 3]
```diff
@@ -83,20 +98,33 @@ export class UsersList extends Component {
...
-                    items={usersList}
-                    Component={User}
+                  {usersList ? (
+                    <Paginate
+                      items={usersList}
+                      Component={User}
+                      pageCount={pageCount}
+                    />
+                  ) : (
+                    'No users found.'
```
Comment (mattwr18) [inline src/containers/UsersList.js:116]: @bryszard do you think you could write a test for this sad path?

**Review by mattwr18 — APPROVED** [review]

**Review by arku — APPROVED** [review]
LGTM :+1:

**Pull request merged**